pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -1736,6 +1737,243 @@ public void testValidateQueryApiError()
+        "SELECT nonExistentColumn FROM mytable", tableConfigs, schemas, null, true);
+
+    String requestJson = JsonUtils.objectToString(request);
+    JsonNode result = JsonUtils.stringToJsonNode(
+        sendPostRequest(getControllerBaseApiUrl() + "/validateMultiStageQuery", requestJson, null));
+    assertFalse(result.get("compiledSuccessfully").asBoolean());
+    assertEquals(result.get("errorCode").asText(), QueryErrorCode.QUERY_VALIDATION.name());
+
+    // Cannot apply '>' to arguments of type '<INTEGER> to <ARRAY>
+    String query = "SELECT DivAirportSeqIDs FROM mytable WHERE DivAirportSeqIDs > 0 LIMIT 10";
+    request = new MultiStageQueryValidationRequest(query, tableConfigs, schemas, null, false);
+
+    requestJson = JsonUtils.objectToString(request);
+    result = JsonUtils.stringToJsonNode(
+        sendPostRequest(getControllerBaseApiUrl() + "/validateMultiStageQuery", requestJson, null));
+
+    assertFalse(result.get("compiledSuccessfully").asBoolean(), "Query should not compile successfully: " + query);
+    assertEquals(result.get("errorCode").asText(), QueryErrorCode.QUERY_VALIDATION.name());
+    assertFalse(result.get("errorMessage").isNull(), "Error message should not be null for: " + query);
+
+    // Non-existent table
+    query = "SELECT count(*) FROM nonExistentTable";
+    request = new MultiStageQueryValidationRequest(query, tableConfigs, schemas, null, false);
+
+    requestJson = JsonUtils.objectToString(request);
+    result = JsonUtils.stringToJsonNode(
+        sendPostRequest(getControllerBaseApiUrl() + "/validateMultiStageQuery", requestJson, null));
+
+    assertFalse(result.get("compiledSuccessfully").asBoolean(), "Query should not compile successfully: " + query);
+    assertEquals(result.get("errorCode").asText(), QueryErrorCode.TABLE_DOES_NOT_EXIST.name());
+    assertFalse(result.get("errorMessage").isNull(), "Error message should not be null for: " + query);
+  }
+
+  @Test
+  public void testValidateQueryApiWithStaticTable()
+      throws Exception {
+    String schemasStr =
+        "{\n" + "  \"schemaName\": \"staticTableTest\",\n" + "  \"enableColumnBasedNullHandling\": false,\n"
+            + "  \"dimensionFieldSpecs\": [\n" + "    {\n" + "      \"name\": \"event_id\",\n"
+            + "      \"dataType\": \"STRING\",\n" + "      \"fieldType\": \"DIMENSION\"\n" + "    },\n" + "    {\n"
+            + "      \"name\": \"dummy_realtime\",\n" + "      \"dataType\": \"STRING\",\n"
+            + "      \"fieldType\": \"DIMENSION\"\n" + "    }\n" + "  ],\n" + "  \"dateTimeFieldSpecs\": [\n"
+            + "    {\n" + "      \"name\": \"mtime\",\n" + "      \"dataType\": \"LONG\",\n"
+            + "      \"fieldType\": \"DATE_TIME\",\n" + "      \"format\": \"1:MILLISECONDS:EPOCH\",\n"
+            + "      \"granularity\": \"1:MILLISECONDS\"\n" + "    }\n" + "  ],\n" + "  \"primaryKeyColumns\": [\n"
+            + "    \"event_id\"\n" + "  ]\n" + "}";
+
+    String tableConfigsStr =
```

Comment:
This makes things very unreadable, can't we use `TableConfigBuilder` and `SchemaBuilder` instead?